plugin/trino-ranger/src/main/java/io/trino/plugin/ranger/RangerSystemAccessControl.java

```diff
@@ -0,0 +1,1236 @@
+    {
+        if (!hasPermission(createResource(schema.getCatalogName(), schema.getSchemaName()), context, RangerTrinoAccessType.DROP, "DropSchema")) {
+            AccessDeniedException.denyDropSchema(schema.getSchemaName());
+        }
+    }
+
+    @Override
+    public void checkCanRenameSchema(SystemSecurityContext context, CatalogSchemaName schema, String newSchemaName)
+    {
+        if (!hasPermission(createResource(schema.getCatalogName(), schema.getSchemaName()), context, RangerTrinoAccessType.ALTER, "RenameSchema")) {
+            AccessDeniedException.denyRenameSchema(schema.getSchemaName(), newSchemaName);
+        }
+    }
+
+    @Override
+    public void checkCanSetSchemaAuthorization(SystemSecurityContext context, CatalogSchemaName schema, TrinoPrincipal principal)
+    {
+        if (!hasPermission(createResource(schema.getCatalogName(), schema.getSchemaName()), context, RangerTrinoAccessType.GRANT, "SetSchemaAuthorization")) {
+            AccessDeniedException.denySetSchemaAuthorization(schema.getSchemaName(), principal);
+        }
+    }
+
+    @Override
+    public void checkCanShowSchemas(SystemSecurityContext context, String catalogName)
+    {
+        if (!hasPermission(createResource(catalogName), context, RangerTrinoAccessType.SHOW, "ShowSchemas")) {
+            AccessDeniedException.denyShowSchemas(catalogName);
+        }
+    }
+
+    @Override
+    public Set<String> filterSchemas(SystemSecurityContext context, String catalogName, Set<String> schemaNames)
+    {
+        Set<String> toExclude = null;
+
+        for (String schemaName : schemaNames) {
+            if (!hasPermissionForFilter(createResource(catalogName, schemaName), context, RangerTrinoAccessType._ANY, "filterSchemas")) {
+                if (toExclude == null) {
+                    toExclude = new HashSet<>();
+                }
+
+                toExclude.add(schemaName);
+            }
+        }
+
+        return toExclude == null ? schemaNames : schemaNames.stream().filter(((Predicate<? super String>) toExclude::contains).negate()).collect(Collectors.toSet());
+    }
+
+    @Override
+    public void checkCanShowCreateSchema(SystemSecurityContext context, CatalogSchemaName schema)
+    {
+        if (!hasPermission(createResource(schema.getCatalogName(), schema.getSchemaName()), context, RangerTrinoAccessType.SHOW, "ShowCreateSchema")) {
+            AccessDeniedException.denyShowCreateSchema(schema.getSchemaName());
+        }
+    }
+
+    @Override
+    public void checkCanCreateTable(SystemSecurityContext context, CatalogSchemaTableName table, Map<String, Object> properties)
+    {
+        if (!hasPermission(createResource(table), context, RangerTrinoAccessType.CREATE, "CreateTable")) {
+            AccessDeniedException.denyCreateTable(table.getSchemaTableName().getTableName());
+        }
+    }
+
+    /**
+     * This is evaluated against the table name as ownership information is not available
```

Comment:
Removed this comment. Owner information, when available, will be useful for authorization - for example, the owner of a table should implicitly be allowed some permissions on the table (like select) without requiring explicit grants. However, AFAIK Trino doesn't not make owner information available to access control implementations. I guess the owner information would be available in the underlying source (like Hive Meta Store); also, this may not be available in all sources that Trino connects to.